hack/generator/pkg/astmodel/identifier_factory.go

```diff
@@ -39,21 +46,37 @@ func NewIdentifierFactory() IdentifierFactory {
 }
 
 // CreateIdentifier returns a valid Go public identifier
-func (factory *identifierFactory) CreateIdentifier(name string) string {
+func (factory *identifierFactory) CreateIdentifier(name string, visibility Visibility) string {
 	if identifier, ok := factory.renames[name]; ok {
-		return identifier
+		name = identifier
```

Comment:
Is the cache really a cache?

Right now the only thing we have in there is `"$schema"` -> `"Schema"` - the issue with returning based on this is that it assumes the cache has knowledge about the desired case. It matters if I want "Schema" or "schema" as the result, and that will depend on if I am looking for a private identifier or a public one?

Maybe I am saying the same as you (the cache is broken as it doesn't take visibility into account) - this change is making it take visibility into account (because now visibility always applies after the renames take place).

Reply:
You're right, it's not really a cache - it's a dictionary of hand crafted transformations, to handle cases where the normal rules don't give the right result.

So the rules we're applying below shouldn't be applied if the dictionary is used, except for the case conversion to handle visibility.

Reply:
Fixed